File: BE/order-service/src/main/java/com/vietnl/orderservice/application/responses/OrderResponse.java
package com.vietnl.orderservice.application.responses;

import com.vietnl.orderservice.domain.models.entities.Order;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Data
public class OrderResponse {

    private UUID id;
    private String tableId;
    private String tableNumber;
    private String createdBy;
    private String status;
    private String note;
    private BigDecimal totalAmount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<OrderItemResponse> items;

    public static OrderResponse from(Order order) {
        OrderResponse r = new OrderResponse();
        r.setId(order.getId());
        r.setTableId(order.getTableId());
        r.setTableNumber(order.getTableNumber());
        r.setCreatedBy(order.getCreatedBy());
        r.setStatus(order.getStatus());
        r.setNote(order.getNote());
        r.setTotalAmount(order.getTotalAmount());
        r.setCreatedAt(order.getCreatedAt());
        r.setUpdatedAt(order.getUpdatedAt());
        r.setItems(
            order.getItems().stream()
                .map(OrderItemResponse::from)
                .collect(Collectors.toList())
        );
        return r;
    }
}

File: FE/mobile/app/src/main/java/com/food/order/data/response/OrderResponse.kt
package com.food.order.data.response

import com.food.order.data.model.Order
import com.food.order.data.model.OrderItem
import com.google.gson.annotations.SerializedName

data class OrderResponse(
    @SerializedName("id") val id: String? = null,
    @SerializedName("tableId") val tableId: String? = null,
    @SerializedName("tableNumber") val tableNumber: String? = null,
    @SerializedName("createdBy") val createdBy: String? = null,
    @SerializedName("status") val status: String? = null,
    @SerializedName("note") val note: String? = null,
    @SerializedName("totalAmount") val totalAmount: Double? = null,
    @SerializedName("createdAt") val createdAt: String? = null,
    @SerializedName("items") val items: List<OrderItem> = emptyList()
) {
    val data: Order
        get() = Order(
            id = id.orEmpty(),
            tableId = tableId,
            createdAt = createdAt,
            createdBy = createdBy,
            status = status,
            items = items,
            totalAmount = totalAmount,
            tableName = tableNumber
        )
}
